File: Linked List/E-206. Reverse Linked List/E-206. Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        reversed_head = None
        while head:
            node = ListNode(head.val, reversed_head)
            reversed_head = node
            head = head.next

        return reversed_head

File: Linked List/E-206. Reverse Linked List/E-206. Reverse Linked List.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */
// head
// 1 -> 2 -> 3 -> 4
// null <- 1
function reverseList(head: ListNode | null): ListNode | null {
  let reversedHead: ListNode | null = null;
  while (head) {
    const node = new ListNode(head.val, reversedHead);
    reversedHead = node;
    head = head.next;
  }
  return reversedHead;
}
